bigint/src/lib.rs

```diff
@@ -1349,6 +1356,47 @@ impl Integer for BigUint {
     }
 }
 
+fn high_bits_to_u64(v: &BigUint) -> u64 {
+    match v.data.len() {
+        0   => 0,
+        1   => v.data[0] as u64,
+        _   => {
+            let mut bits = v.bits();
+            let mut ret = 0u64;
+            let mut ret_bits = 0;
+
+            for d in v.data.iter().rev() {
+                let digit_bits = (bits + big_digit::BITS - 1) %
+                    big_digit::BITS + 1;
```

Comment:
Yes.  On 32-bit platforms, a `BigUint` that grows more than 512MB is going to have a bad time anywhere that's using `::bits()`.  One that's just on the edge could create the possible-but-unlikely addition overflow here.  Meh.